src/components/dialog/dialog.js

```diff
@@ -213,6 +259,11 @@ class Dialog extends Modal {
     if (!this.props.title) { return null; }
 
     let title = this.props.title;
+    const classes = classNames(
+          'carbon-dialog__title', {
+            'carbon-dialog__title--has-subheader': this.props.subtitle
+          }
+        );
```

Comment:
Im still not convinced but there is no need for this to block the PR. If the linter is happy